exams/pearson.py

```diff
@@ -154,14 +223,17 @@ def upload_tsv(file_path):
                 "The {} setting is required".format(key)
             )
 
-    cnopts = pysftp.CnOpts()
-    cnopts.hostkeys = None  # ignore knownhosts
-    with pysftp.Connection(
-        host=str(settings.EXAMS_SFTP_HOST),
-        port=int(settings.EXAMS_SFTP_PORT),
-        username=str(settings.EXAMS_SFTP_USERNAME),
-        password=str(settings.EXAMS_SFTP_PASSWORD),
-        cnopts=cnopts,
-    ) as sftp:
-        with sftp.cd(settings.EXAMS_SFTP_UPLOAD_DIR):
-            sftp.put(file_path)
+    try:
+        cnopts = pysftp.CnOpts()
+        cnopts.hostkeys = None  # ignore knownhosts
+        with pysftp.Connection(
+            host=str(settings.EXAMS_SFTP_HOST),
+            port=int(settings.EXAMS_SFTP_PORT),
+            username=str(settings.EXAMS_SFTP_USERNAME),
+            password=str(settings.EXAMS_SFTP_PASSWORD),
+            cnopts=cnopts,
+        ) as sftp:
+            with sftp.cd(settings.EXAMS_SFTP_UPLOAD_DIR):
+                sftp.put(file_path)
+    except (ConnectionException, SSHException) as ex:
+        raise RetryableSFTPException() from ex
```

Comment:
please, try to wrap only the statements for which you want to handle exceptions in a `try/except`, even if you have to write a `try/except` for each statement. 
in case of problems it will be much easier to find the origin of the failure.